### DISCUSSION TITLE
Windows Terminal v1.20.11271.0

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#17200] [Announcements]
It feels like just a few days ago that 1.20 got updated in the preview channel! How strange...

Here are the highlights:
* The new rendering engine is enabled by default for everyone!
* We've rewritten resizing so that it's faster and more reliable.
* Alternate Scroll Mode is on by default, so you can scroll pagers like `less` with the mouse wheel.

> [!IMPORTANT]
> The new rendering engine is enabled by default, and the original one will be **removed** in 1.21.
> Please report any issues you encounter with it.

Please see the following release notes for additional details.

* [Windows Terminal Preview v1.20.11215.0](https://github.com/microsoft/terminal/releases/tag/v1.20.11215.0)
* [Windows Terminal Preview v1.20.10822.0](https://github.com/microsoft/terminal/releases/tag/v1.20.10822.0)
* [Windows Terminal Preview v1.20.10572.0](https://github.com/microsoft/terminal/releases/tag/v1.20.10572.0)
* [Windows Terminal Preview v1.20.10303.0](https://github.com/microsoft/terminal/releases/tag/v1.20.10303.0)
* [Windows Terminal Preview v1.20.10293.0](https://github.com/microsoft/terminal/releases/tag/v1.20.10293.0)

Note that scrollbar marks and shell integration features are still considered experimental and
_are not included in this stable release!_ However, they are slated for inclusion in 1.21 when it is released to the stable channel.

In addition, automatic search highlighting has been disabled for this release.

> [!NOTE]
> This version began rolling out to the Dev, Canary and Beta Windows Insider channels on 2024-05-07. Other channels will be released
> as our reliability numbers indicate that we haven't broken anything. If you want to update to this release directly, download it below!

### Why are there so many files? How do I choose?

Please visit our page documenting the different [Windows Terminal Distributions]!

## Bug Fixes

_The stable release of Terminal also comes with the following bugfixes backported from 1.21._

* Double-clicking a word on the left side of the screen will no longer select parts of your `RPROMPT` (or right prompt) (#17170)

### Reliability

* The Atlas engine has gotten somewhat better at reporting errors (#17193)

_With additional thanks to @jvr2022 for keeping our dependencies up to date._

[Windows Terminal Distributions]: https://learn.microsoft.com/windows/terminal/distributions

---

_Binary files inside the unpackaged distribution archive bear the version number `1.20.240506001`._

### Asset Hashes

- Microsoft.WindowsTerminal_1.20.11271.0_8wekyb3d8bbwe.msixbundle_Windows10_PreinstallKit.zip
   - SHA256 `4BFB5E319CE068AA286C97B5D181837A6E453B16DAFC9A4F4D006DF15FB76083`
- Microsoft.WindowsTerminal_1.20.11271.0_8wekyb3d8bbwe.msixbundle
   - SHA256 `67216C17DDAB0BAF93E1BE16A851C8D904351DA497FB3C1CD2B71DF849B37931`
- Microsoft.WindowsTerminal_1.20.11271.0_arm64.zip
   - SHA256 `0177B1A17D06150C6A6723DA3AF74C2B42A668DC7DF87C594C3C07CC104FB66B`
- Microsoft.WindowsTerminal_1.20.11271.0_x64.zip
   - SHA256 `F480F65A7874E7055C51EC67E0E4F98011A4EF63DD43C6B882458DAE36A83286`
- Microsoft.WindowsTerminal_1.20.11271.0_x86.zip
   - SHA256 `05DB4E20EB6718858D1C10C0A2042372E9B9C8AB83385DAF389164C03C0E3E20`

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.20.11271.0'>Windows Terminal v1.20.11271.0</a>.</em>

**Comment by stanio** [comment]
> The new rendering engine is enabled by default for everyone!

Does this refer to the "AtlasEngine text renderer"?  After installing the new Terminal version it hadn't been enabled for me but then I've previously enabled and disabled it back explicitly.  Enabling it again now I'm still seeing a character gap with [block characters](https://en.wikipedia.org/wiki/Block_Elements) used for progress bars:

![renderer-atlas](https://github.com/microsoft/terminal/assets/1247730/62ed1541-f7fa-4e78-8771-de23797d0b56)

Here's a reference with the AtlasEngine disabled (smooth/continuous progress bar):

![renderer-default](https://github.com/microsoft/terminal/assets/1247730/99c81303-9ce4-478f-b3bc-13bfc63967ea)

This is on Windows 11, screen scaling 200%, CaskaydiaCove [Nerd Font](https://github.com/ryanoasis/nerd-fonts).  Funny enough I'm not currently seeing the same (the character gaps with AtlasEngine enabled) on my Windows 10 laptop with screen scaling 200%, while I've previously seen it there as well.

**Reply by stanio** [reply]
>   The new rendering engine is enabled by default, and the original one will be **removed** in 1.21.
>   Please report any issues you encounter with it.

Just tried Terminal Preview v1.21.1272.0 – not seeing the gaps:

![terminal-1.21](https://github.com/microsoft/terminal/assets/1247730/09452893-c90d-4956-b806-5814a12ffbaf)

So it is probably fine.

**Reply by lhecker** [reply]
v1.21 contains this PR which fixes the gaps: #16729

**Comment by andreihes** [comment]
what abt `command paletter -> nt...`? why its gone?

**Reply by DHowett** [reply]
Oh no! I'm sorry about that. We didn't consider that some of the UI stylistic changes would impact muscle memory :/

**Reply by zadjii-msft** [reply]
oh yea I didn't even thing about that. I kind of hate it now. 

Actually wait we probably should entirely revert that for the commands. Because now the (formerly) "New tab..." (&pane) command names conflict with the un-nested ones, so there's _no_ nested New Tab item at all anymore. Yikes.

**Reply by andreihes** [reply]
I didn't get the muscule memory reference. New tab menu does not exist anymore anywhere. There is no way to open a profile in a new tab using any kind of filtering / search / etc. I had to scroll the 100+ items list to find one I need thus just got back to v1.18...

**Comment by trueuto** [comment]
Why is Store so badly behind the release schedule? I know it takes some time, but it is 3 weeks now and in Store there is still the 1.19.11213.0 version available.
